src/libos/src/net/syscalls.rs

```diff
@@ -416,6 +416,10 @@ pub async fn do_setsockopt(
         fd, level, optname, optval, optlen
     );
 
+    if optlen == 0 {
```

Comment:
I think the optlen should be 0 when optval is NULL.

Reply:
Actually in current impelmentation, the optval NULL  will return EFAULT in `is_inside_user_space check`.

Reply:
What I think is, any code should try to explain a clear logic by itself. In this case, "If the optval is NULL, then the error code would be set to EFAULT in some place. Maybe it is correct (I am not sure). But it is not a clear truth for every reader."